axlearn/common/utils.py

```diff
@@ -1158,6 +1159,63 @@ def prune_tree(
     return in_tree
 
 
+def tree_merge_default_leaf_merge_fn(primary: Any, secondary: Any):
```

Comment:
The behavior of this function still seem too complicated to be the default function for tree_merge. Can we ask the user to specify the merge function explicitly?

Reply:
Done.